base/libgit2/oid.jl

```diff
@@ -51,32 +61,40 @@ function GitHash(repo::GitRepo, ref_name::AbstractString)
     return oid_ptr[]
 end
 
-function GitHash(obj::Ptr{Void})
-    oid_ptr = ccall((:git_object_id, :libgit2), Ptr{UInt8}, (Ptr{Void},), obj)
-    oid_ptr == C_NULL && return GitHash()
-    return GitHash(oid_ptr)
-end
-
-function GitHash{T<:GitObject}(obj::T)
-    obj === nothing && return GitHash()
-    return GitHash(obj.ptr)
+function GitHash(obj::GitObject)
+    GitHash(ccall((:git_object_id, :libgit2), Ptr{UInt8}, (Ptr{Void},), obj.ptr))
 end
 
 Base.hex(id::GitHash) = join([hex(i,2) for i in id.val])
+Base.hex(id::GitShortHash) = hex(id.hash)[1:id.len]
```

Comment:
It's only used in a handful of functions: `git_oid_ncmp` and `git_object_lookup_prefix`